src/_pytest/nodes.py

```diff
@@ -144,8 +144,10 @@ def __init__(
                 self._nodeid += "::" + self.name
 
     @classmethod
-    def from_parent(cls, parent, *, name):
-        return cls._create(parent=parent, name=name)
+    def from_parent(cls, parent, **kw):
+        assert "config" not in kw
```

Comment:
This is public API, so we really should raise `TypeError` here for the forbidden parameters.